in_toto/model.go

```diff
@@ -220,16 +341,58 @@ func (mb *Metablock) Load(path string) error {
 
 	if signed["_type"] == "link" {
 		var link Link
-		if err := json.Unmarshal(*rawMb["signed"], &link); err != nil {
+		reflection := reflect.TypeOf(link)
+		attributeCount := reflection.NumField()
+		allFields := make([]string, 0)
+
+		for i := 0; i < attributeCount; i++ {
+			allFields = append(allFields, reflection.Field(i).Tag.Get("json"))
+		}
+
+		for _, field := range allFields {
+			if _, ok := signed[field]; !ok {
+				return fmt.Errorf("required field %s missing", field)
+			}
+		}
```

Comment:
IMO It should be enough to just call `validateLink` after having deserialized the json file. It should blow up if the any of the validated link's fields are invalid, which includes required fields being  empty (`nil`?).

Reply:
Same goes for `validateLayout` below.

Reply:
You're suggesting moving the required field check to `validateLink` and `validateLayout` respectively? I think in the meeting we had last month to discuss validation, we planned to check for the fields before unmarshalling into the structure? I may be confused. It should be an easy fix though.

Reply:
We only need to validate before unmarshalling, to catch undefined fields, since those would be silently discarded during unmarshalling (you already do this with `DisallowUnknownFields`).

But I don't see the benefit of also checking required fields upfront, if they are validated afterwards anyway. Or am I missing something? 

Here's a simple example
```go
type Foo struct {
	Bar string `json:"bar"`
}
func validateFoo(foo Foo) error {
	if foo.Bar != "this string" {
		return fmt.Errorf("invalid foo: Bar must be 'this string', got: %s", foo.Bar)
	}
}
```

Now if you call `validateFoo` after unmashalling any json into `Foo`, it will blow up if the json was not `{"bar": "this string"}`, including the case where `"bar"` wasn't even in the json.

Reply:
@adityasaky, just informed me that this approach won't work, because the json unmarshaller initializes missing fields to the type's default value ("", 0, False ...), which might be a valid value for a required field.

https://github.com/golang/go/issues/17163 has some workarounds and what @adityasaky is doing is actually even better. So please disregard my comment from above.

However, it might be worth to create a function that does the json attribute iterating, to not repeat it for links and layouts...